sonic-net/sonic-buildimage
# [Build] Use apt-get to predictably support dependency ordered configuration of lazy packages

#### Why I did it

- The current lazy installer relies on a filename sort for both unpack and configuration steps. When systemd services are configured [started] by multiple packages the order is by filename not by the declared package dependencies.   This can cause the start order of services to differ between first-boot and subsequent boots.   Declared systemd service dependencies further exacerbate the issue (e.g. blocking the first-boot script).

- The current installer leaves packages un-configured if the package dependency order does not match the filename order.

- This also fixes a trivial bug in #10923 where externally downloaded dependencies are duplicated across lazy package device directories.

#### How I did it
Changed the staging and first-boot scripts to use apt-get:

`dpkg -i /host/image-$SONIC_VERSION/platform/$platform/*.deb`

becomes

`apt-get -y install /host/image-$SONIC_VERSION/platform/$platform/*.deb`

when dependencies are detected during image staging.


#### How to verify it
Apt-get  [critical rules](https://github.com/Debian/apt/blob/main/apt-pkg/orderlist.cc)

Add a Depends= to the control information of a package.  Grep the syslog for rc.local between images and observe the configuration order of packages change.

#### Which release branch to backport (provide reason below if selected)

- [ X] 202012
- [ X] 202205

#### Description for the changelog

Use apt-get to predictably support dependency ordered configuration of lazy packages
